docs/core/diagnostics/diagnostic-libraries.md

```diff
@@ -0,0 +1,161 @@
+---
+title: Diagnostic libraries
+description: An overview of the available diagnostic libraries in .NET.
+ms.date: 10/16/2023
+ms.topic: overview
+---
+
+# Diagnostic libraries
+
+Diagnostic libraries are tools to collect and analyze diagnostic information about your .NET applications. From summarizing exceptions, determining resource utilization and monitoring, to exposing health checks, and providing probes for Kubernetes. The diagnostic packages provide a rich set of functionality that can be used to instrument your apps.
+
+The following NuGet packages are available:
```

Comment:
There are other diagnostic libraries supplied in .NET, those listed here happen to be the new ones. I think the title and description of this article are thus misleading, and perhaps this whole article should be split.

There is already content in the .NET docs about health checks, so I think the health-related stuff should move there. https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/health-checks?view=aspnetcore-7.0

The health probe topic is about reporting the process health to the environment, so that the environment can stop routing traffic to the process when it is unhealthy. This should probably go into ASP.NET docs.

The exception summarization is used in telemetry (mainly for metrics). So it should go into a telemetry-related area.

Reply:
> There are other diagnostic libraries supplied in .NET, those listed here happen to be the new ones. I think the title and description of this article are thus misleading, and I think this whole article should be split since it combines things together which don't belong together.

I'm not opposed to the splitting of content, could you please help direct me as to where you think each bit of content fits within the .NET docs? Here is a link to the current .NET Diagnostics content: https://learn.microsoft.com/dotnet/core/diagnostics, as I believe it should be added somewhere within this subset of content.

> There is already content in the .NET docs about health checks, so I think the health-related stuff should move there. https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/health-checks?view=aspnetcore-7.0

The link you provided is the ASP.NET Core docs, not .NET docs. We are workload agnostic. We can link to that content for additional reading as it relates to ASP.NET Core (or web-specific workloads). Also, their content touches on the primitives, but exposes this functionality through middleware — a package that sits atop these two health check packages.

> The health probe topic is about reporting the process health to the environment, so that the environment can stop routing traffic to the process when it is unhealthy. This should probably go into ASP.NET docs.

It seems like this library isn't ready yet anyway, perhaps we exclude it for now? Also, it's specific to Kubernetes, no? I don't think of that being synonymous with ASP.NET docs, as that is workload agnostic.

Reply:
I split the content and attempted to put it where I thought it made the most sense.